In [1]:
import numpy as no
import pandas as pd
import os
from pathlib import Path
import swifter

In [2]:
os.getcwd()

'd:\\GitHub\\MSBD5005_DataVizProject'

In [3]:
base_path = (Path(os.getcwd()) / 'Datasets/Activity Logs').resolve()
all_activity_log_file_name = os.listdir(base_path)

In [4]:
raw_file = []
for file in all_activity_log_file_name:
    if '.csv' in file:
        print(file)
        raw_file.append(pd.read_csv(base_path / file))

ParticipantStatusLogs1.csv
ParticipantStatusLogs10.csv
ParticipantStatusLogs11.csv
ParticipantStatusLogs12.csv
ParticipantStatusLogs13.csv
ParticipantStatusLogs14.csv
ParticipantStatusLogs15.csv
ParticipantStatusLogs16.csv
ParticipantStatusLogs17.csv
ParticipantStatusLogs18.csv
ParticipantStatusLogs19.csv
ParticipantStatusLogs2.csv
ParticipantStatusLogs20.csv
ParticipantStatusLogs21.csv
ParticipantStatusLogs22.csv
ParticipantStatusLogs23.csv
ParticipantStatusLogs24.csv
ParticipantStatusLogs25.csv
ParticipantStatusLogs26.csv
ParticipantStatusLogs27.csv
ParticipantStatusLogs28.csv
ParticipantStatusLogs29.csv
ParticipantStatusLogs3.csv
ParticipantStatusLogs30.csv
ParticipantStatusLogs31.csv
ParticipantStatusLogs32.csv
ParticipantStatusLogs33.csv
ParticipantStatusLogs34.csv
ParticipantStatusLogs35.csv
ParticipantStatusLogs36.csv
ParticipantStatusLogs37.csv
ParticipantStatusLogs38.csv
ParticipantStatusLogs39.csv
ParticipantStatusLogs4.csv
ParticipantStatusLogs40.csv
ParticipantStatusLogs41.

In [5]:
raw_file = pd.concat(raw_file, axis = 0)


In [6]:
raw_file.shape

(113923735, 12)

In [7]:
raw_file = raw_file.reset_index(drop=True)

In [10]:
# type(raw_file['timestamp'][0])
raw_file['timestamp'] = pd.to_datetime( raw_file['timestamp'], format="%Y-%m-%dT%H:%M:%SZ")

In [11]:
raw_file.head()

timestamp                                 currentLocation  participantId  \
0 2022-03-01  POINT (-2724.6277665310454 6866.2081834436985)            0.0   
1 2022-03-01  POINT (-1526.9372331431534 5582.2951345645315)            1.0   
2 2022-03-01   POINT (-1360.9905987829304 2108.804385379679)            2.0   
3 2022-03-01    POINT (-1558.517200825967 5600.664347152427)            3.0   
4 2022-03-01     POINT (976.2409614204214 4574.575079082071)            4.0   

  currentMode hungerStatus sleepStatus  apartmentId  availableBalance  jobId  \
0      AtHome      JustAte    Sleeping        926.0       1286.519556  254.0   
1      AtHome      JustAte    Sleeping        928.0        860.574204  929.0   
2      AtHome      JustAte    Sleeping        291.0       1298.184541  348.0   
3      AtHome      JustAte    Sleeping       1243.0       1180.641725  316.0   
4      AtHome      JustAte    Sleeping        194.0       -681.650588  177.0   

  financialStatus  dailyFoodBudget  weeklyExtraBudget  
0          Stable             12.0        1104.302570  
1          Stable             12.0         926.714377  
2          Stable             16.0         848.802876  
3          Stable             12.0         819.325405  
4        Unstable             20.0           0.000000

In [17]:
raw_file['date'] = raw_file['timestamp'].dt.date
# raw_file.iloc[:3,:]['timestamp'].dt.date

In [18]:
daily_pid_jid = raw_file[['date', 'participantId', 'jobId']].drop_duplicates()
daily_pid_jid['date'].value_counts().plot(kind ='line')

In [38]:
daily_jid = daily_pid_jid[['date', 'jobId']]

In [48]:
daily_jid = daily_jid.merge(right=job_df[['jobId', 'employerId']], on=['jobId'], how='left')

In [51]:
daily_jid.head()

date  jobId  employerId
0  2022-03-01  254.0       424.0
1  2022-03-01  929.0      1321.0
2  2022-03-01  348.0       440.0
3  2022-03-01  316.0       436.0
4  2022-03-01  177.0       411.0

In [79]:
att_base_path = base_path.parent / 'Attributes'
job_df = pd.read_csv(att_base_path / 'Jobs.csv')
employer_df = pd.read_csv(att_base_path / 'Employers.csv')

There are toally 1192 Jobs Id and 254 Employers. 

Job ID range: [1 .. 1326] (1325)

Employer ID range: [379 .. 1797] (1419)

In [92]:
pubs_df = pd.read_csv(att_base_path / 'Pubs.csv')
food_df = pd.read_csv(att_base_path / 'Restaurants.csv')
school_df = pd.read_csv(att_base_path / 'Schools.csv')

In [93]:
pubs_df

pubId  hourlyCost  maxOccupancy  \
0     442    8.281103            64   
1     443    6.417435            64   
2     444   12.581806            84   
3     892   11.642905            96   
4     893   14.840473            79   
5     894   12.311210            72   
6    1342   14.140037            77   
7    1343   10.524972            67   
8    1344   10.415293            60   
9    1798   11.545351            64   
10   1799    7.487082            73   
11   1800   10.206845            62   

                                         location  buildingId  
0     POINT (964.4380231713202 3991.603473784208)         556  
1     POINT (1809.880173357865 4339.172426035451)          29  
2     POINT (770.4279044387976 932.5852003214752)        1012  
3   POINT (-1524.9573211662105 3815.271490114369)         502  
4   POINT (-1608.766411449925 3886.4924784954583)         164  
5   POINT (-1702.5378155541193 3865.477090848479)         238  
6   POINT (-260.45747053139064 5026.150804179361)         429  
7   POINT (-501.42676062779447 3551.572638649489)         489  
8    POINT (-225.9209866450221 3878.053697626113)         585  
9    POINT (-3209.738139187618 6487.657688065099)         953  
10  POINT (-3903.194075780787 5967.8369837734845)         234  
11  POINT (-3796.1198513891923 6367.922072582076)         627

In [94]:
food_df

restaurantId  foodCost  maxOccupancy   \
0            445      5.15             71   
1            446      4.17             82   
2            447      5.87            119   
3            448      4.07             98   
4            449      5.11             53   
5            895      4.44            103   
6            896      5.86            107   
7            897      5.76            104   
8            898      4.22            115   
9            899      5.65             85   
10          1345      4.70             72   
11          1346      5.85             74   
12          1347      4.41             87   
13          1348      4.23             57   
14          1349      4.77            104   
15          1801      5.51             64   
16          1802      5.08             72   
17          1803      4.35             94   
18          1804      5.92             48   
19          1805      5.59             85   

                                          location  buildingId  
0     POINT (631.5130723031391 2001.4772026036535)         304  
1       POINT (413.840000705876 1194.128694228948)         308  
2      POINT (497.9967937001494 1624.515148185587)          58  
3      POINT (698.2411158717262 4392.416668183332)         964  
4    POINT (1407.7107695149243 4010.4574815269225)         181  
5   POINT (-1623.0740591441258 3872.0300113645017)         164  
6    POINT (-2126.1723943111892 4285.182228265561)         619  
7      POINT (-1989.63526350385 3368.764837974986)         875  
8     POINT (-1771.452216635636 4343.580176583138)         917  
9     POINT (-820.9290279479601 4497.243308530136)          86  
10  POINT (-1176.8013302032396 5846.6471242364805)         991  
11    POINT (-3529.558386975454 4809.641127666926)          27  
12       POINT (-1221.4051100732 4941.00338419236)         679  
13   POINT (-508.12691044634744 6307.584421447018)         124  
14   POINT (-1176.6989055164295 4726.117600229471)         888  
15   POINT (-3136.1278958093117 6671.917461867151)         160  
16   POINT (-4002.9335312988205 6671.996711133368)         714  
17    POINT (-3182.885940731813 7167.608699410916)         213  
18     POINT (-4131.41409222454 7236.525633225643)         101  
19    POINT (-3927.9102395808727 5494.63962663673)         285